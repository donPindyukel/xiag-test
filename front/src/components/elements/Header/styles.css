.header-wrap {
  display: flex;
  flex-direction: column;

  .header {
    display: flex;
    justify-content: space-between;
    padding: 20px 20px 20px 40px;
    @media (width < 768px) {
      align-items: center;
    }
    @media (width < 421px) {
      padding-left: 20px;
    }

    .header-back-link {
      a {
        display: block;
        width: 135px;
        height: 42px;
        background: url(../../../images/page-logo.png) center no-repeat;
        @media (width < 421px) {
          background-size: contain;
          width: 90px;
          margin-right: 20px;
        }
      }
    }

    .title {
      color: var(--headerFontColor);
      font-size: var(--headerFontSize);
      font-weight: 300;
      line-height: 39px;
      @media (width < 768px) {
        font-size: 14px;
        line-height: 1.2;
        min-width: 162px;
      }
    }
  }

  .header-title {
    background: url(../../../images/header.jpg) center 0 no-repeat;
    height: 95px;
    line-height: 95px;
    background-size: cover;
    text-align: center;

    .title {
      display: inline-block;
      vertical-align: middle;
      background-color: rgba(0, 0, 0, 0.2);
      color: var(--cleanWhite);
      font-size: 50px;
      font-weight: 300;
      line-height: 39px;
      padding: 5px 30px;
      @media (width < 768px) {
        font-size: 20px;
      }
    }
  }
}
