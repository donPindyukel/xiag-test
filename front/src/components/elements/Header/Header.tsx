import React from "react";
import "./styles.css";

export const Header: React.SFC = () => {
  return (
    <div className="header-wrap">
      <div className="header">
        <div className="header-back-link">
          <a href="https://www.xiag.ch" target="_blank" />
        </div>
        <div className="title">Poll website task: HTML example</div>
      </div>
      <div className="header-title">
        <div className="title">Poll website task: HTML example</div>
      </div>
    </div>
  );
};
