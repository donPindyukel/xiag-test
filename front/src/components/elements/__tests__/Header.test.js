import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Header } from "../Header/Header";

describe("<Header />", () => {
  let component;

  describe("Header", () => {
    beforeAll(() => {
      component = shallow(<Header />);
    });

    it("renders the component", () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});