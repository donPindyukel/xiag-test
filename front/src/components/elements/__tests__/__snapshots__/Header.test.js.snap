// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Header renders the component 1`] = `
<div
  className="header-wrap"
>
  <div
    className="header"
  >
    <div
      className="header-back-link"
    >
      <a
        href="https://www.xiag.ch"
        target="_blank"
      />
    </div>
    <div
      className="title"
    >
      Poll website task: HTML example
    </div>
  </div>
  <div
    className="header-title"
  >
    <div
      className="title"
    >
      Poll website task: HTML example
    </div>
  </div>
</div>
`;
